### ISSUE TITLE
Raises Error with Facebook API, but gives a valid access token.

### ISSUE THREAD

**Issue by bwheeler96** [oauth-xx/oauth2#145]
## The Error

Basically I run the user through the oauth2 flow, direct them to the sign in page, my app is _successfully_ granted permissions, and then the user is redirected by facebook to my callback action, and when I use

`oauth_consumer_object.auth_code.get_token(params[:code], :redirect_uri => 'http://localhost:3000')`

OAuth2 throws an `OAuth2::Error`, because the response is not in the format it wants.
## The cause

If we trace back the method `get_token` we will eventually find a line that says

`rails (Error) if options[:raise_error] &&  ! (response['parsed'].is_a? Hash)`

or something like that.
## Bottom line

A _valid access token_ is returned, but OAuth2 still raises an error.

**Comment by josephpage** [comment]
The solution was found in #75.
